### PR TITLE
Use containerd client library to connect to containerd services

### PR DESCRIPTION
**Pull request by mikebrow**
Switching to using the containerd client library for service interfaces

Signed-off-by: Mike Brown <brownwm@us.ibm.com>

### CODE REVIEW

**Comment by Random-Liu** [comment]
@mikebrow Why don't you `godep save ./...` to include the dependency?

**Comment by mikebrow** [comment]
I ran godep restore to make sure I'm on the right version of everything.. that checks out just fine. 

I then ran godep save to verify it's working fine with no changes to vendor or godeps.json etc. That checked out fine.

I added the import to "github.com/containerd/containerd" in services.go

I get it to build and verify.. make && make verify.  All fine..

I then I run godep save ./... and it adds github.com/containerd/containerd and also deletes my vendor/github.com/containerd/containerd/api/* etc. files.

So what the heck.. godep can't handle the containerd package at the root of containerd/containerd and the sub packages under neath it at the same time.

**Comment by Random-Liu** [comment]
@mikebrow Mine works fine.

```
$ # Add new dependency
$ godep restore
$ remove -rf vendor
$ remove -rf Godeps
$ godep save ./...
```

**Comment by mikebrow** [comment]
debugging the cri-tools verification test not having much luck

**Comment by mikebrow** [comment]
Thx for the help @Random-Liu should be ready to review now

**Comment by mikebrow** [comment]
CRI validation test results:
```
Summarizing 6 Failures:

[Fail] [k8s.io] Networking runtime should support networking [It] runtime should support port mapping with host port and container port [Conformance] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/networking.go:206

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support portforward [Conformance] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/streaming.go:231

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support attach [Conformance] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/streaming.go:183

[Fail] [k8s.io] Streaming runtime should support streaming interfaces [It] runtime should support exec [Conformance] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/streaming.go:127

[Fail] [k8s.io] Security Context runtime should support container with security context [It] runtime should support RunAsUserName [security context] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/container.go:326

[Fail] [k8s.io] Security Context runtime should support container with security context [It] runtime should support RunAsUser [security context] 
/home/mike/go/src/github.com/kubernetes-incubator/cri-tools/pkg/validate/container.go:326

Ran 36 of 36 Specs in 565.895 seconds
FAIL! -- 30 Passed | 6 Failed | 0 Pending | 0 Skipped --- FAIL: TestE2ECRI (565.90s)
FAIL

Ginkgo ran 1 suite in 9m25.953293741s
Test Suite Failed
exit status 1
```

File: pkg/server/service.go [hunk 1]
```diff
@@ -99,11 +100,19 @@ type criContainerdService struct {
...
 }
 
 // NewCRIContainerdService returns a new instance of CRIContainerdService
-func NewCRIContainerdService(conn *grpc.ClientConn, rootDir, networkPluginBinDir, networkPluginConfDir string) (CRIContainerdService, error) {
+func NewCRIContainerdService(conn *grpc.ClientConn, containerdEndpoint, rootDir, networkPluginBinDir, networkPluginConfDir string) (CRIContainerdService, error) {
 	// TODO(random-liu): [P2] Recover from runtime state and metadata store.
+
+	client, err := containerd.New(containerdEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize client: %v", err)
```
Comment (Random-Liu) [inline pkg/server/service.go]: nit: failed to initialize containerd client with endpoint %q: %v

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -7,6 +7,11 @@
 	],
 	"Deps": [
 		{
+			"ImportPath": "github.com/Microsoft/go-winio",
+			"Comment": "v0.3.8",
```
Comment (Random-Liu) [inline Godeps/Godeps.json]: Change to `v0.4.1`.
https://github.com/containerd/containerd/blob/8ed1e24ae925b5c6d8195858ee89dddb0507d65f/vendor.conf#L35

File: pkg/server/service.go [hunk 3]
```diff
@@ -105,10 +97,10 @@ type criContainerdService struct {
 }
 
 // NewCRIContainerdService returns a new instance of CRIContainerdService
-func NewCRIContainerdService(conn *grpc.ClientConn, containerdEndpoint, rootDir, networkPluginBinDir, networkPluginConfDir string) (CRIContainerdService, error) {
+func NewCRIContainerdService(containerdEndpoint, rootDir, networkPluginBinDir, networkPluginConfDir string) (CRIContainerdService, error) {
 	// TODO(random-liu): [P2] Recover from runtime state and metadata store.
 
-	client, err := containerd.New(containerdEndpoint)
```
Comment (Random-Liu) [inline pkg/server/service.go]: Move `k8sContainerdNamespace` to this file? Since we use it in here now.

**Comment by Random-Liu** [comment]
LGTM overall with several nits. Could you squash the commits? At least squash the vetshadows change into one.

File: pkg/server/service.go (hunk 1 above)
Comment (mikebrow) [inline pkg/server/service.go]: done

File: pkg/server/service.go (hunk 3 above)
Comment (mikebrow) [inline pkg/server/service.go]: done

File: Godeps/Godeps.json (hunk 2 above)
Comment (mikebrow) [inline Godeps/Godeps.json]: done

File: pkg/server/service.go [hunk 4]
```diff
@@ -39,6 +39,11 @@ import (
 	"github.com/kubernetes-incubator/cri-containerd/pkg/server/agents"
 )
 
+const (
```
Comment (Random-Liu) [inline pkg/server/service.go]: const k8sContainerdNamespace = "k8s.io"

since we only have one const now.

File: pkg/server/server.go [hunk 5]
```diff
@@ -31,8 +31,6 @@ import (
 const (
 	// unixProtocol is the network protocol of unix socket.
 	unixProtocol = "unix"
```
Comment (Random-Liu) [inline pkg/server/server.go]: Same here.

**Comment by Random-Liu** [comment]
@mikebrow Please squash your commits, at least get rid of comment address and revert commits.

**Comment by Random-Liu** [comment]
LGTM

**Comment by Random-Liu** [comment]
Will merge after test passes.

**Pull request merged**